apps/api/src/app/plugins/bffCache.ts

```diff
@@ -49,43 +54,83 @@ export const bffCache: FastifyPluginCallback<BffCacheOptions> = (
         reply.header(HEADER_NAME, 'HIT');
         reply.type('application/json');
         reply.send(item);
-        return;
+        fastify.log.debug('onRequest cacheItem: %s (%d)', item, ttlInSeconds);
+
+        return reply;
       }
+    } else {
+      fastify.log.trace(`Request not cached for "${key}`);
+      // For now we don't add the cache 'MISS' header, because we only do this for cacheable endpoints (the ones that include cache-control header in the response)
+      // We need to handle this in the "onSend" once we know the actual response
     }
-
-    return;
   });
 
   fastify.addHook('onSend', async function (req, reply, payload) {
-    const isCacheHit = reply.getHeader(HEADER_NAME) === 'HIT';
-    const cacheTtl: number | undefined = getTtlFromResponse(reply, ttl);
-    const isStatus200 = reply.statusCode >= 200 && reply.statusCode < 300;
+    try {
+      const isCacheHit = reply.getHeader(HEADER_NAME) === 'HIT';
+      const cacheTtl: number | undefined = getTtlFromResponse(reply, ttl);
+      const isStatus200 = reply.statusCode >= 200 && reply.statusCode < 300;
+
+      // If the cache is a hit, or is non-cacheable, we just proceed with the request
+      if (isCacheHit || cacheTtl === undefined || !isStatus200) {
+        return undefined;
```

Comment:
If this is a cache hit, shouldn't it reply with `payload` instead of `undefined`?